dev/CommonStyles/APITests/CommonStylesTests.cs

```diff
@@ -34,6 +36,137 @@ namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests
+        }
+
+        bool ResourceDictionariesContainSameKeys(ResourceDictionary expectedDictionary, string expectedDictionaryName, ResourceDictionary actualDictionary, string actualDictionaryName)
+        {
+            List<string> missingKeysInActualDictionary = new List<string>();
+            foreach (var entry in expectedDictionary)
+            {
+                if (!actualDictionary.ContainsKey(entry.Key))
+                {
+                    missingKeysInActualDictionary.Add(entry.Key.ToString());
+                }
+            }
+
+            if (missingKeysInActualDictionary.Count > 0)
+            {
+                Log.Comment("Keys found in " + expectedDictionaryName + " but not in " + actualDictionaryName);
+                foreach (var missingKey in missingKeysInActualDictionary)
+                {
+                    Log.Comment("* " + missingKey);
+                }
+            }
+
+            return (missingKeysInActualDictionary.Count == 0);
+        }
+
+        [TestMethod]
+        public void VerifyNoBaselineResourceWasRemoved()
+        {
+            var actualResourcesKeys = new HashSet<string>();
+
+            RunOnUIThread.Execute(() =>
+            {
+                var resourceDictionaries = new XamlControlsResources() { ControlsResourcesVersion = ControlsResourcesVersion.Version2 };
+
+                foreach (var dictionaryName in resourceDictionaries.ThemeDictionaries.Keys)
+                {
+                    var themeDictionary = resourceDictionaries.ThemeDictionaries[dictionaryName] as ResourceDictionary;
+
+                    foreach (var entry in themeDictionary)
+                    {
+                        string entryKey = entry.Key as string;
+                        if (!actualResourcesKeys.Contains(entryKey))
+                        {
+                            actualResourcesKeys.Add(entryKey);
+                        }
+                    }
+                }
+
+                foreach (var entry in resourceDictionaries)
+                {
+                    string entryKey = entry.Key as string;
+                    if (!actualResourcesKeys.Contains(entryKey))
+                    {
+                        actualResourcesKeys.Add(entryKey);
+                    }
+                }
+            });
+
+            StringBuilder missingKeysList = new StringBuilder();
+
+            bool allBaselineResourceKeysExist = true;
+            foreach (var baselineResourceKey in BaselineResources.BaselineResourcesList)
+            {
+                if (!actualResourcesKeys.Contains(baselineResourceKey))
+                {
+                    missingKeysList.Append(baselineResourceKey + ", ");
+                    allBaselineResourceKeysExist = false;
+                }
+            }
+
+            Verify.IsTrue(allBaselineResourceKeysExist, "List of missing resource keys: " + missingKeysList.ToString());
```

Comment:
Please provide a more detailed message like "You have removed a key that was previously shipped. We could have apps that have taken a dependency on this key which will crash due to that. If you do not need to use this key in the styles anymore, please move it into the deprcitead_themeresources.xaml file.